File: Models/PlugSets/DestinyPlugSetDefinition.cs
namespace bulkybook.Models;

public class DestinyPlugSetDefinition
{
    public DestinyDisplayPropertiesDefinition displayProperties {get;set;}
    public List<DestinyItemSocketEntryPlugItemRandomizedDefinition> reusablePlugItems {get;set;}
    public Boolean isFakePlugSet {get;set;}
    public UInt32 hash {get;set;}
    public Int32 index {get;set;}
    public Boolean redacted {get;set;}
}   

File: Scripts/Models/DestinyPlugSetDefinition.ts
import { DestinyDisplayPropertiesDefinition } from "./DestinyDisplayPropertiesDefinition";
import { DestinyItemSocketEntryPlugItemRandomizedDefinition } from "./DestinyItemSocketEntryPlugItemRandomizedDefinition";
export interface DestinyPlugSetDefinition
{
    displayProperties: DestinyDisplayPropertiesDefinition;
    reusablePlugItems: Array<DestinyItemSocketEntryPlugItemRandomizedDefinition>;
    isFakePlugSet: boolean;
    hash: number;
    index: number;
    redacted: boolean;
}